pkg/replica/replica.go

```diff
@@ -709,38 +706,53 @@ func (r *Replica) createNewHead(oldHead, parent, created string, size int64) (f
 		UserCreated: false,
 		Created:     created,
 	}
-	err = r.encodeToFile(&newDisk, newHeadName+diskutil.DiskMetadataSuffix)
-	return f, newDisk, err
+	subRollbackFunc, err = r.encodeToFile(&newDisk, newHeadName+diskutil.DiskMetadataSuffix)
```

Comment:
As I mentioned above, what `subRollbackFun` will do has been done in the defer function of `encodeToFile`, so you should use the returned error.

```go
types.CombineErrors(errorHappenedBeforeDefer, r.rmDisk(newHeadName))
```

Reply:
Clarified by @shuo-wu

Reply:
Oh Yes. Here `subRollbackFunc` will be executed twice if this `r.encodeToFile` failed...
I am thinking we can set `rollbackFunc` to `nil` when the original function errors out. Then we don't need to worry about the duplicate execution issued by the upper layer.

Reply:
Sounds good to me.